pytensor/tensor/rewriting/basic.py

```diff
@@ -320,56 +321,52 @@ def dimshuffled_alloc(i):
     return new_outs
 
 
-@register_canonicalize("shape_unsafe")
 @node_rewriter([Elemwise])
 def local_fill_sink(fgraph, node):
     """
     f(fill(a, b), fill(c, d), e) -> fill(c, fill(a, f(b, d, e)))
     f need to be an elemwise that isn't a fill.
     """
-    if not hasattr(node, "op") or not isinstance(node.op, Elemwise) or node.op == fill:
+    if isinstance(node.op.scalar_op, Second):
```

Comment:
The extra checks were only needed for the recursive call of this rewrite. A default call will never call it on an node without Op that is not already Elemwise